compiler/rustc_borrowck/src/diagnostics/conflict_errors.rs

```diff
@@ -613,7 +613,12 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
                     if self.sugg_span.is_some() {
                         return;
                     }
-                    if let hir::StmtKind::Local(hir::Local { span, ty, init: None, .. }) = &ex.kind
+
+                    // FIXME: We make sure that this is a normal binding and not a destructuring assignment,
```

Comment:
this is not destructuring assignment, it's just a normal binding that's not at the top level